Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/COREX12AuditLogger.java

```diff
@@ -0,0 +1,216 @@
+        if (auditLogMsg != null && auditLogMsg.getAuditMessage() != null) {
+            audit(auditLogMsg, assertion);
+        } else {
+            LOG.error("Core X12 Realtime Respponse auditLogMsg is null");
+        }
+        LOG.trace("---End COREX12AuditLogger.auditCoreX12RealtimeRespponse()---");
+    }
+
+    /**
+     *
+     * @param message
+     * @param assertion
+     * @param target
+     * @param direction
+     */
+    public void auditNhinCoreX12BatchRequest(COREEnvelopeBatchSubmission message, AssertionType assertion, NhinTargetSystemType target,
+        String direction, boolean isRequesting) {
+        LOG.trace("---Begin COREX12AuditLogger.auditNhinCoreX12BatchRequest()---");
+        LogEventRequestType auditLogMsg = logNhinCoreX12BatchRequest(message, assertion, target, direction, isRequesting);
+        if (auditLogMsg != null && auditLogMsg.getAuditMessage() != null) {
+            audit(auditLogMsg, assertion);
+        } else {
+            LOG.error("Core X12 Nhin Batch Request auditLogMsg is null");
+        }
+        LOG.trace("---End COREX12AuditLogger.auditNhinCoreX12BatchRequest()---");
+    }
+
+    /**
+     *
+     * @param message
+     * @param assertion
+     * @param target
+     * @param direction
+     * @param isRequesting
+     */
+    public void auditNhinCoreX12BatchResponse(COREEnvelopeBatchSubmissionResponse message, AssertionType assertion, NhinTargetSystemType target,
+        String direction, boolean isRequesting) {
+        LOG.trace("---Begin COREX12AuditLogger.auditNhinCoreX12BatchResponse()---");
+        LogEventRequestType auditLogMsg = logNhinCoreX12BatchResponse(message, assertion, target, direction, NhincConstants.AUDIT_LOG_NHIN_INTERFACE, isRequesting);
+        if (auditLogMsg != null && auditLogMsg.getAuditMessage() != null) {
+            audit(auditLogMsg, assertion);
+        } else {
+            LOG.error("Core X12 Nhin Batch Response auditLogMsg is null");
+        }
+        LOG.trace("---End COREX12AuditLogger.auditNhinCoreX12BatchResponse()---");
+    }
+
+    /**
+     * Submits a generic Audit Log message to the Audit Log Repository.
+     *
+     * @param auditLogMsg The generic audit log to be audited
+     * @param assertion The assertion to be audited
+     * @return
+     */
+    private AcknowledgementType audit(LogEventRequestType auditLogMsg, AssertionType assertion) {
+        AuditRepositoryProxyObjectFactory auditRepoFactory = new AuditRepositoryProxyObjectFactory();
+        AuditRepositoryProxy proxy = auditRepoFactory.getAuditRepositoryProxy();
+        return proxy.auditLog(auditLogMsg, assertion);
+    }
+    
+        /**
+     *
+     * @param message
+     * @param assertion
+     * @param target
+     * @param direction
+     * @param _interface
+     * @return LogEventRequestType
+     */
+    public LogEventRequestType logNhinCoreX12RealtimeRequest(COREEnvelopeRealTimeRequest message, AssertionType assertion, NhinTargetSystemType target,
```

Comment:
Is there any reason for this to be its own function?  'transformRequestToAuditMsg' has its own logging, so this function really boils down to a single return.  Strongly consider removing this function altogether.